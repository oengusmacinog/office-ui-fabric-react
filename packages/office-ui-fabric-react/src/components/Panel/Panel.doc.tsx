import * as React from 'react';
import { PanelSmallRightExample } from './examples/Panel.SmallRight.Example';

import { IDocPageProps } from '../../common/DocPage.types';
import { PanelSmallLeftExample } from './examples/Panel.SmallLeft.Example';
import { PanelSmallFluidExample } from './examples/Panel.SmallFluid.Example';
import { PanelMediumExample } from './examples/Panel.Medium.Example';
import { PanelLargeExample } from './examples/Panel.Large.Example';
import { PanelLargeFixedExample } from './examples/Panel.LargeFixed.Example';
import { PanelExtraLargeExample } from './examples/Panel.ExtraLarge.Example';
import { PanelCustomExample } from './examples/Panel.Custom.Example';
import { PanelHiddenOnDismissExample } from './examples/Panel.HiddenOnDismiss.Example';
import { PanelLightDismissExample } from './examples/Panel.LightDismiss.Example';
import { PanelLightDismissCustomExample } from './examples/Panel.LightDismissCustom.Example';
import { PanelNonModalExample } from './examples/Panel.NonModal.Example';
import { PanelFooterExample } from './examples/Panel.Footer.Example';
import { PanelStatus } from './Panel.checklist';

const PanelSmallRightExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.SmallRight.Example.tsx') as string;
const PanelSmallRightExampleCodepen = require('!raw-loader!office-ui-fabric-react/lib/codepen/components/Panel/Panel.SmallRight.Example.Codepen.txt') as string;
const PanelSmallLeftExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.SmallLeft.Example.tsx') as string;
const PanelSmallFluidExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.SmallFluid.Example.tsx') as string;
const PanelMediumExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.Medium.Example.tsx') as string;
const PanelLargeExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.Large.Example.tsx') as string;
const PanelLargeFixedExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.LargeFixed.Example.tsx') as string;
const PanelExtraLargeExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.ExtraLarge.Example.tsx') as string;
const PanelCustomExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.Custom.Example.tsx') as string;
const PanelHiddenOnDismissExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.HiddenOnDismiss.Example.tsx') as string;
const PanelLightDismissExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.LightDismiss.Example.tsx') as string;
const PanelLightDismissCustomExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.LightDismissCustom.Example.tsx') as string;
const PanelNonModalExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.NonModal.Example.tsx') as string;
const PanelFooterExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Panel/examples/Panel.Footer.Example.tsx') as string;

export const PanelPageProps: IDocPageProps = {
  title: 'Panel',
  componentName: 'Panel',
  componentUrl:
    'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/Panel',
  componentStatus: PanelStatus,
  examples: [
    {
      title: 'Panel - Small Panel, Anchored Right, Fixed Width',
      code: PanelSmallRightExampleCode,
      view: <PanelSmallRightExample />,
      codepenJS: PanelSmallRightExampleCodepen
    },
    {
      title: 'Panel - Small Panel, Anchored Left, Fixed Width',
      code: PanelSmallLeftExampleCode,
      view: <PanelSmallLeftExample />
    },
    {
      title: 'Panel - Small Panel, Full Screen, Fluid Width',
      code: PanelSmallFluidExampleCode,
      view: <PanelSmallFluidExample />
    },
    {
      title: 'Panel - Medium',
      code: PanelMediumExampleCode,
      view: <PanelMediumExample />
    },
    {
      title: 'Panel - Large',
      code: PanelLargeExampleCode,
      view: <PanelLargeExample />
    },
    {
      title: 'Panel - LargeFixed',
      code: PanelLargeFixedExampleCode,
      view: <PanelLargeFixedExample />
    },
    {
      title: 'Panel - Extra Large',
      code: PanelExtraLargeExampleCode,
      view: <PanelExtraLargeExample />
    },
    {
      title: 'Panel - Custom',
      code: PanelCustomExampleCode,
      view: <PanelCustomExample />
    },
    {
      title: 'Panel - Hidden on Dismiss',
      code: PanelHiddenOnDismissExampleCode,
      view: <PanelHiddenOnDismissExample />
    },
    {
      title: 'Panel - Light Dismiss',
      code: PanelLightDismissExampleCode,
      view: <PanelLightDismissExample />
    },
    {
      title: 'Panel - Custom Light Dismiss',
      code: PanelLightDismissCustomExampleCode,
      view: <PanelLightDismissCustomExample />
    },
    {
      title: 'Panel - Non-Modal',
      code: PanelNonModalExampleCode,
      view: <PanelNonModalExample />
    },
    {
      title: 'Panel - Footer',
      code: PanelFooterExampleCode,
      view: <PanelFooterExample />
    }
  ],
  propertiesTablesSources: [require<string>('!raw-loader!office-ui-fabric-react/src/components/Panel/Panel.types.ts')],
  overview: require<string>('!raw-loader!office-ui-fabric-react/src/components/Panel/docs/PanelOverview.md'),
  bestPractices: '',
  dos: require<string>('!raw-loader!office-ui-fabric-react/src/components/Panel/docs/PanelDos.md'),
  donts: require<string>('!raw-loader!office-ui-fabric-react/src/components/Panel/docs/PanelDonts.md'),
  isHeaderVisible: true,
  isFeedbackVisible: true,
  allowNativeProps: true
};
