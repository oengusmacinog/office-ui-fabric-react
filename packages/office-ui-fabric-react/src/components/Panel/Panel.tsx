/* tslint:disable:no-unused-variable */
import * as React from 'react';
/* tslint:enable:no-unused-variable */

import {
  BaseComponent,
  autobind,
  css,
  getId,
  getRTL
} from '../../Utilities';
import { FocusTrapZone } from '../FocusTrapZone/index';
import { IPanel, IPanelProps, PanelType } from './Panel.types';
import { Layer } from '../Layer/Layer';
import { Overlay } from '../../Overlay';
import { Popup } from '../../Popup';
import { IconButton } from '../../Button';
import { AnimationClassNames, getTheme, FontSizes, IconFontSizes } from '../../Styling';
import * as stylesImport from './Panel.scss';
const styles: any = stylesImport;
const theme = getTheme();

export interface IPanelState {
  isFooterSticky?: boolean;
  isOpen?: boolean;
  isAnimating?: boolean;
  id?: string;
}

export class Panel extends BaseComponent<IPanelProps, IPanelState> implements IPanel {

  public static defaultProps: IPanelProps = {
    isHiddenOnDismiss: false,
    isOpen: false,
    isBlocking: true,
    hasCloseButton: true,
    type: PanelType.smallFixedFar,
  };

  private _content: HTMLElement;

  constructor(props: IPanelProps) {
    super(props);

    this._warnDeprecations({
      'ignoreExternalFocusing': 'focusTrapZoneProps',
      'forceFocusInsideTrap': 'focusTrapZoneProps',
      'firstFocusableSelector': 'focusTrapZoneProps'
    });

    this.state = {
      isFooterSticky: false,
      isOpen: false,
      isAnimating: false,
      id: getId('Panel')
    };
  }

  public componentDidMount() {
    this._events.on(window, 'resize', this._updateFooterPosition);

    if (this.props.isOpen) {
      this.open();
    }
  }

  public componentWillReceiveProps(newProps: IPanelProps) {
    if (newProps.isOpen !== this.state.isOpen) {
      if (newProps.isOpen) {
        this.open();
      } else {
        this.dismiss();
      }
    }
  }

  public render() {
    const {
      className = '',
      elementToFocusOnDismiss,
      firstFocusableSelector,
      focusTrapZoneProps,
      forceFocusInsideTrap,
      hasCloseButton,
      headerText,
      ignoreExternalFocusing,
      isBlocking,
      isLightDismiss,
      isHiddenOnDismiss,
      layerProps,
      type,
      customWidth,
      onLightDismissClick = this._onPanelClick,
      onRenderNavigation = this._onRenderNavigation,
      onRenderHeader = this._onRenderHeader,
      onRenderBody = this._onRenderBody,
      onRenderFooter = this._onRenderFooter
    } = this.props;
    const { isOpen, isAnimating, id } = this.state;
    const isLeft = type === PanelType.smallFixedNear ? true : false;
    const isRTL = getRTL();
    const isOnRightSide = isRTL ? isLeft : !isLeft;
    const headerTextId = id + '-headerText';
    const customWidthStyles = (type === PanelType.custom) ? { width: customWidth } : {};
    const renderProps: IPanelProps = { ...this.props, componentId: id };

    if (!isOpen && !isAnimating && !isHiddenOnDismiss) {
      return null;
    }

    let overlay;
    if (isBlocking && isOpen) {
      overlay = (
        <Overlay
          className={ css(
            styles.overlay,
            isOpen && isAnimating && AnimationClassNames.fadeIn200,
            !isOpen && isAnimating && AnimationClassNames.fadeOut200
          ) }
          isDarkThemed={ false }
          onClick={ isLightDismiss ? onLightDismissClick : undefined }
        />
      );
    }

    return (
      <Layer { ...layerProps }>
        <Popup
          role='dialog'
          ariaLabelledBy={ headerText && headerTextId }
          onDismiss={ this.dismiss }
          className={
            css(
              !isOpen && !isAnimating && isHiddenOnDismiss && styles.hiddenPanel
            )
          }
        >
          <div
            className={
              css(
                'ms-Panel',
                styles.root,
                className,
                // because the RTL animations are not being used, we need to set a class
                isOpen && ('is-open ' + styles.rootIsOpen),
                type === PanelType.smallFluid && ('ms-Panel--smFluid ' + styles.rootIsSmallFluid),
                type === PanelType.smallFixedNear && ('ms-Panel--smLeft ' + styles.rootIsSmallLeft),
                type === PanelType.smallFixedFar && ('ms-Panel--sm ' + styles.rootIsSmall),
                type === PanelType.medium && ('ms-Panel--md ' + styles.rootIsMedium),
                (type === PanelType.large || type === PanelType.largeFixed) && ('ms-Panel--lg ' + styles.rootIsLarge),
                type === PanelType.largeFixed && ('ms-Panel--fixed ' + styles.rootIsFixed),
                type === PanelType.extraLarge && ('ms-Panel--xl ' + styles.rootIsXLarge),
                type === PanelType.custom && ('ms-Panel--custom ' + styles.rootIsCustom),
                hasCloseButton && ('ms-Panel--hasCloseButton ' + styles.rootHasCloseButton),
                !isOpen && !isAnimating && isHiddenOnDismiss && styles.hiddenPanel
              )
            }
          >
            { overlay }
            <FocusTrapZone
              ignoreExternalFocusing={ ignoreExternalFocusing }
              forceFocusInsideTrap={ forceFocusInsideTrap }
              firstFocusableSelector={ firstFocusableSelector }
              { ...focusTrapZoneProps }
              className={
                css(
                  'ms-Panel-main',
                  styles.main,
                  isOpen && isAnimating && !isOnRightSide && AnimationClassNames.slideRightIn40,
                  isOpen && isAnimating && isOnRightSide && AnimationClassNames.slideLeftIn40,
                  !isOpen && isAnimating && !isOnRightSide && AnimationClassNames.slideLeftOut40,
                  !isOpen && isAnimating && isOnRightSide && AnimationClassNames.slideRightOut40,
                  focusTrapZoneProps ? focusTrapZoneProps.className : undefined
                ) }
              style={ customWidthStyles }
              elementToFocusOnDismiss={ elementToFocusOnDismiss }
              isClickableOutsideFocusTrap={ isLightDismiss || isHiddenOnDismiss }
            >
              <div className={ css('ms-Panel-commands') } data-is-visible={ true } >
                { onRenderNavigation(renderProps, this._onRenderNavigation) }
              </div>
              <div className={ css('ms-Panel-contentInner', styles.contentInner) } >
                { onRenderHeader(renderProps, this._onRenderHeader) }
                { onRenderBody(renderProps, this._onRenderBody) }
                { onRenderFooter(renderProps, this._onRenderFooter) }
              </div>
            </FocusTrapZone>
          </div>
        </Popup>
      </Layer>
    );
  }

  public open() {
    if (!this.state.isOpen) {
      this.setState({
        isOpen: true,
        isAnimating: true
      }, () => {
        this._async.setTimeout(this._onTransitionComplete, 200);
      });
    }
  }

  @autobind
  public dismiss() {
    if (this.state.isOpen) {
      this.setState({
        isOpen: false,
        isAnimating: true
      }, () => {
        this._async.setTimeout(this._onTransitionComplete, 200);
      });

      if (this.props.onDismiss) {
        this.props.onDismiss();
      }
    }
  }

  @autobind
  private _onRenderNavigation(props: IPanelProps): JSX.Element | null {
    const {
      closeButtonAriaLabel,
      hasCloseButton
    } = props;
    if (hasCloseButton) {
      return (
        <div className={ css('ms-Panel-navigation', styles.navigation) } >
          <IconButton
            styles={
              {
                root: {
                  height: 'auto',
                  width: '44px',
                  color: theme.palette.neutralSecondary,
                  fontSize: IconFontSizes.large,
                },
                rootHovered: {
                  color: theme.palette.neutralPrimary
                }
              }
            }
            className={ css('ms-Panel-closeButton ms-PanelAction-close') }
            onClick={ this._onPanelClick }
            ariaLabel={ closeButtonAriaLabel }
            data-is-visible={ true }
            iconProps={ { iconName: 'Cancel' } }
          />
        </div>
      );
    }
    return null;
  }

  @autobind
  private _onRenderHeader(props: IPanelProps): JSX.Element | null {
    const {
      headerText,
      componentId,
      headerClassName = '',
    } = props;

    if (headerText) {
      return (
        <div className={ css('ms-Panel-header', styles.header) }>
          <p className={ css('ms-Panel-headerText', styles.headerText, headerClassName) } id={ componentId + '-headerText' } role='heading'>
            { headerText }
          </p>
        </div>
      );
    }
    return null;
  }

  @autobind
  private _onRenderBody(props: IPanelProps): JSX.Element {
    const contentClass = css(
      'ms-Panel-content',
      styles.content,
      props.isFooterAtBottom && styles.contentGrow
    );

    return (
      <div ref={ this._resolveRef('_content') } className={ contentClass } >
        { props.children }
      </div>
    );
  }

  @autobind
  private _onRenderFooter(props: IPanelProps): JSX.Element | null {
    const { isFooterSticky } = this.state;
    const { onRenderFooterContent = null } = this.props;
    if (onRenderFooterContent) {
      return (
        <div className={ css('ms-Panel-footer', styles.footer, isFooterSticky && styles.footerIsSticky) } >
          <div className={ css('ms-Panel-footerInner', styles.footerInner) }>
            { onRenderFooterContent() }
          </div>
        </div>
      );
    }
    return null;
  }

  private _updateFooterPosition() {
    const _content = this._content;
    if (_content) {
      const height = _content.clientHeight;
      const innerHeight = _content.scrollHeight;

      this.setState({
        isFooterSticky: height < innerHeight ? true : false
      });
    }
  }

  @autobind
  private _onPanelClick() {
    this.dismiss();
  }

  @autobind
  private _onTransitionComplete() {
    this.setState({
      isAnimating: false
    });

    if (!this.state.isOpen && this.props.onDismissed) {
      this.props.onDismissed();
    }
  }
}
