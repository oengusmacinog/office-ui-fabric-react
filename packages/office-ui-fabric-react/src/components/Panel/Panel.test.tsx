import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';

import { PanelBase } from './Panel.base';
import { Panel } from './Panel';
let div: HTMLElement;

// Mock Layer since it otherwise shows nothing in snapshot tests
jest.mock('../../Layer', () => {
  return {
    Layer: jest.fn().mockImplementation(props => {
      return props.children;
    })
  };
});

describe('Panel', () => {
  it('renders Panel correctly', () => {
    const component = renderer.create(
      <Panel isOpen={true} headerText="Test Panel">
        <span>Content goes here</span>
      </Panel>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('onClose', () => {
    beforeEach(() => {
      div = document.createElement('div');
    });

    afterEach(() => {
      ReactDOM.unmountComponentAtNode(div);
    });

    it('fires the correct events when closing', done => {
      let dismissedCalled = false;
      let dismissCalled = false;
      const setDismissTrue = (): void => {
        dismissCalled = true;
      };
      const setDismissedTrue = (): void => {
        dismissedCalled = true;
      };

      const panel: PanelBase = ReactDOM.render(
        <PanelBase isOpen={true} onDismiss={setDismissTrue} onDismissed={setDismissedTrue} />,
        div
      ) as any;

      panel.dismiss();

      expect(dismissCalled).toEqual(true);
      expect(dismissedCalled).toEqual(false);

      setTimeout(() => {
        try {
          expect(dismissedCalled).toEqual(true);
          done();
        } catch (e) {
          done(e);
        }
      }, 250);
    });
  });
});
