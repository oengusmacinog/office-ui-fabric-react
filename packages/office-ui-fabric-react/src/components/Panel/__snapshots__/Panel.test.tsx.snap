// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders Panel correctly 1`] = `
<div
  aria-describedby={undefined}
  aria-label={undefined}
  aria-labelledby="Panel0-headerText"
  className=


  onKeyDown={[Function]}
  role="dialog"
  style={
    Object {
      "overflowY": undefined,
    }
  }
>
  <div
    className=
        ms-Panel
        is-open
        ms-Panel--hasCloseButton
        {
          bottom: 0px;
          left: 0px;
          pointer-events: none;
          position: absolute;
          right: 0px;
          top: 0px;
        }
    type={1}
  >
    <div
      className=
          ms-Overlay
          {
            animation-duration: 0.267s;
            animation-fill-mode: both;
            animation-name: keyframes from{opacity:0;}to{opacity:1;};
            animation-timing-function: cubic-bezier(.1,.25,.75,.9);
            background-color: rgba(255,255,255,.4);
            bottom: 0px;
            cursor: pointer;
            left: 0px;
            opacity: 1;
            pointer-events: auto;
            position: absolute;
            right: 0px;
            top: 0px;
            transition: opacity 0.367s cubic-bezier(.1,.9,.2,1);
          }
          @media screen and (-ms-high-contrast: active){& {
            border: 1px solid WindowText;
          }
      onClick={undefined}
    />
    <div
      aria-labelledby={undefined}
      className=
          ms-Panel-main
          {
            -webkit-overflow-scrolling: touch;
            animation-duration: 0.367s;
            animation-fill-mode: both;
            animation-name: keyframes from{opacity:0;}to{opacity:1;} keyframes from{transform:translate3d(40px,0,0);}to{transform:translate3d(0,0,0);};
            animation-timing-function: cubic-bezier(.1,.9,.2,1);
            background-color: #ffffff;
            bottom: 0px;
            overflow-x: hidden;
            overflow-y: auto;
            position: absolute;
            right: 0px;
            top: 0px;
            width: 100%;
          }
          @media (min-width: 480px){& {
            border-left: 1px solid #eaeaea;
            border-right: 1px solid #eaeaea;
            box-shadow: -30px 0px 30px -30px rgba(0,0,0,0.2);
            left: auto;
            pointer-events: auto;
            width: 340px;
          }
          $root & {
            pointer-events: auto;
            width: 272px;
          }
          @media (min-width: 480px){$root & {
            width: 340px;
          }
      onFocusCapture={[Function]}
      onKeyDown={[Function]}
      style={Object {}}
    >
      <div
        className=
            ms-Panel-scrollableContent
            {
              height: 100%;
            }
      >
        <div
          className=
              ms-Panel-commands

          data-is-visible={true}
        >
          <div
            className=
                ms-Panel-navigation
                {
                  display: flex;
                  height: 44px;
                  justify-content: flex-end;
                  padding-bottom: 0;
                  padding-left: 5px;
                  padding-right: 5px;
                  padding-top: 0;
                }
          >
            <button
              aria-describedby={undefined}
              aria-disabled={undefined}
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-pressed={undefined}
              className=
                  ms-Button
                  ms-Button--icon
                  ms-Panel-closeButton
                  ms-PanelAction-close
                  {
                    -moz-osx-font-smoothing: grayscale;
                    -webkit-font-smoothing: antialiased;
                    background-color: transparent;
                    border-radius: 0px;
                    border: 1px solid transparent;
                    box-sizing: border-box;
                    color: #666666;
                    cursor: pointer;
                    display: inline-block;
                    font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                    font-size: 20px;
                    font-weight: 400;
                    height: auto;
                    outline: transparent;
                    padding-bottom: 0;
                    padding-left: 4px;
                    padding-right: 4px;
                    padding-top: 0;
                    position: relative;
                    text-align: center;
                    text-decoration: none;
                    user-select: none;
                    vertical-align: top;
                    width: 44px;
                  }
                  &::-moz-focus-inner {
                    border: 0;
                  }
                  .ms-Fabric--isFocusVisible &:focus:after {
                    border: 1px solid #ffffff;
                    bottom: 0px;
                    content: "";
                    left: 0px;
                    outline: 1px solid #666666;
                    position: absolute;
                    right: 0px;
                    top: 0px;
                    z-index: 1;
                  }
                  @media screen and (-ms-high-contrast: active){.ms-Fabric--isFocusVisible &:focus:after {
                    border: none;
                    bottom: -2px;
                    left: -2px;
                    outline-color: ButtonText;
                    right: -2px;
                    top: -2px;
                  }
                  &:hover {
                    color: #333333;
                  }
                  @media screen and (-ms-high-contrast: active){&:hover {
                    border-color: Highlight;
                    color: Highlight;
                  }
                  &:active {
                    color: #0078d4;
                  }
              data-is-focusable={true}
              data-is-visible={true}
              disabled={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              type="button"
            >
              <div
                className=
                    ms-Button-flexContainer
                    {
                      align-items: center;
                      display: flex;
                      flex-wrap: nowrap;
                      height: 100%;
                      justify-content: center;
                    }
              >
                <i
                  aria-hidden={true}
                  className=
                      ms-Button-icon
                      {
                        -moz-osx-font-smoothing: grayscale;
                        -webkit-font-smoothing: antialiased;
                        display: inline-block;
                        flex-shrink: 0;
                        font-family: "FabricMDL2Icons";
                        font-size: 16px;
                        font-style: normal;
                        font-weight: normal;
                        height: 16px;
                        line-height: 16px;
                        margin-bottom: 0;
                        margin-left: 4px;
                        margin-right: 4px;
                        margin-top: 0;
                        speak: none;
                        text-align: center;
                        vertical-align: middle;
                      }
                  data-icon-name="Cancel"
                  role="presentation"
                >
                  îœ‘
                </i>
              </div>
            </button>
          </div>
        </div>
        <div
          className=
              ms-Panel-contentInner
              {
                -webkit-overflow-scrolling: touch;
                display: flex;
                flex-direction: column;
                min-height: calc(100% - 44px);
                top: 44px;
                transform: translateZ(0);
              }
        >
          <div
            className=
                ms-Panel-header
                {
                  flex-grow: 0;
                  margin-bottom: 14px;
                  margin-left: 0;
                  margin-right: 0;
                  margin-top: 14px;
                  padding-left: 16px;
                  padding-right: 16px;
                }
                @media screen and (min-width: 768px){& {
                  padding-left: 32px;
                  padding-right: 32px;
                }
                @media screen and (min-width: 1366px){& {
                  padding-left: 40px;
                  padding-right: 40px;
                }
                @media (min-width: 1024px){& {
                  margin-top: 30px;
                }
          >
            <p
              aria-level={2}
              className=
                  ms-Panel-headerText
                  {
                    -moz-osx-font-smoothing: grayscale;
                    -webkit-font-smoothing: antialiased;
                    color: #333333;
                    font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                    font-size: 21px;
                    font-weight: 100;
                    line-height: 32px;
                    margin-bottom: 0px;
                    margin-left: 0px;
                    margin-right: 0px;
                    margin-top: 0px;
                  }
              id="Panel0-headerText"
              role="heading"
            >
              Test Panel
            </p>
          </div>
          <div
            className=
                ms-Panel-content
                {
                  margin-bottom: 0px;
                  overflow-y: auto;
                  padding-bottom: 20px;
                  padding-left: 16px;
                  padding-right: 16px;
                }
                @media screen and (min-width: 768px){& {
                  padding-left: 32px;
                  padding-right: 32px;
                }
                @media screen and (min-width: 1366px){& {
                  padding-left: 40px;
                  padding-right: 40px;
                }
            data-is-scrollable={true}
          >
            <span>
              Content goes here
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
